imu_integrator/plugin.cpp

```diff
@@ -0,0 +1,477 @@
+#include <chrono>
+#include <iomanip>
+#include <thread>
+#include <eigen3/Eigen/Dense>
+
+#include "../open_vins/ov_msckf/src/state/Propagator.h"
+#include "../open_vins/ov_core/src/types/IMU.h"
```

Comment:
I don't like components depending on other components instead of `common/`. What if we switch to OrbSLAM? I understand we will have to change a lot of code in OrbSLAM, but it would be a better starting if `imu_integrator` doesn't hard-depend on OpenVINS. However, I understand this might be necessary here.

Is it possible to make a struct in `common/` that gets included in OpenVINS and in `imu_integrator` that does this job?

Reply:
I agree, having a generic integrator component depend on some other component/library's code seems hard to maintain and definitely tangles things